packages/ckeditor5-engine/src/model/documentfragment.ts

```diff
@@ -23,7 +35,13 @@ import isIterable from '@ckeditor/ckeditor5-utils/src/isiterable';
  * will be set to the {@link module:engine/model/model~Model#markers model markers} by a
  * {@link module:engine/model/writer~Writer#insert} function.
  */
-export default class DocumentFragment {
+export default class DocumentFragment implements Iterable<Node> {
+	public readonly markers: Map<string, Range>;
+	declare public name?: undefined;
+	declare public rootName?: undefined;
```

Comment:
Maybe those should be marked as internal?

Reply:
Nope. They are for the cases when you have something like this:

```typescript
let parent: Element | DocumentFragment;
if ( parent.name == 'xyz' /* would fail if DocumentFragment didn't have `name` property */ ) {
	DoSomething( parent ); // it know that parent is Element, because on DocumentFragment it's undefined and can't equal 'xyz'
}
```

Reply:
Nope. They are for the cases when you have something like this:

```typescript
let parent: Element | DocumentFragment;
if ( parent.name == 'xyz' /* would fail if DocumentFragment didn't have `name` property */ ) {
	DoSomething( parent ); // it know that parent is Element, because on DocumentFragment it's undefined and can't equal 'xyz'
}
```

Reply:
And on `Node` they're public